app/src/org/commcare/android/storage/framework/Persisted.java

```diff
@@ -98,6 +97,7 @@ public int getID() {
     }
 
     private void readVal(Field f, Object o, DataInputStream in) throws DeserializationException, IOException, IllegalAccessException {
+        synchronized (fieldOrderings) {
```

Comment:
Updated and updated on https://github.com/dimagi/commcare-odk/pull/1304 where I also ran the tests for the fix.